### DISCUSSION TITLE
how to use the `on_click` properly (README isn't clear enough)

### DISCUSSION THREAD

**Discussion by daUnknownCoder** [nvim-lualine/lualine.nvim#1259] [Configuration]
hey devs, nice job on adding the `on_click`, its awesome but it isnt documented properly (a few examples in the wiki will be good enough)

I figured it out *(half part) thanks to @miversen33 in his reddit comment: [Bro's Comment](https://new.reddit.com/r/neovim/comments/w4gcek/comment/ii9sjp2/?context=3)

I wanted to ask about the main thing thats left: the mouse button thing (i want different funcs for left press and right press)

What i have done till now:
```lua
            on_click = function()
              vim.cmd(":LspInfo")
            end,
```
i want something like:
```lua
            on_click = function(button)
              if button == "left" then
                vim.cmd(":LspInfo")
              end
              if button == "right" then
                vim.cmd(":LspRestart")
              end
            end,
```

which is not working/im dumb enough to not understand this

**Comment by miversen33** [comment]
It's been a hot minute since I have dicked with my lualine config. I might poke this tonight and see if I can get you some answers :) I did get onclick implemented on the aforementioned plugin so it shouldn't be too hard for me to puzzle through how to use the updated stuff. I didn't know they added all that new stuff, that's pretty awesome :)

**Comment by iton0** [comment]
This should work:
```lua
            on_click = function(_, button, _)
              if button == "l" then
                vim.cmd(":LspInfo")
              end
              if button == "r" then
                vim.cmd(":LspRestart")
              end
            end,
```
